### DISCUSSION TITLE
Allure reports loading forever

### DISCUSSION THREAD

**Discussion by KristofKuli** [allure-framework/allure2#2368] [Bug Reports] [answered]
Dear Support,

The generated Allure reports are more than 100MB in size and after successful publishing, we can't see them as it loads forever. Is there any size limitation for Allure? We have checked the network logs and this is what we can see:
```
ColumnNumber: 67893
​
fileName: "https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js"
​
lineNumber: 1
​
message: "u.value[0] is undefined"
​
name: "TypeError"
​
stack: "740/n</</</</<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:67893\n740/p</u/</<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:66894\n740/p</u/<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:66999\ns@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65692\npromise callback*a@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65857\ns@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65689\npromise callback*a@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65857\ns@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65689\npromise callback*a@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65857\ns@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65689\npromise callback*a@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65857\ns@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65689\npromise callback*a@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65857\n740/a</<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65868\n740/a<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:65635\n740/n</</<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:67180\n@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:59951\n584/r</e.prototype.invokeMethod@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:56930\n584/r</e.prototype.onMessage@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:57647\n584/r</e/this._handleMessageReceived@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:60930\nEventListener.handleEvent*e@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:61175\n584/r<@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:61445\n584@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:64884\nr@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:76599\n740@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:67017\nr@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:76599\n@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:76967\n@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:76975\n@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:158\n@https://qameta.gallerycdn.vsassets.io/extensions/qameta/allure-azure-pipelines/1.0.6/1707134962806/dist/main.js:1:223\n"
​
toString: function i()​
<prototype>: Object { … }
```

![image](https://github.com/allure-framework/allure2/assets/103577855/3c737fab-a3cf-4f9d-add9-d4398f6dec7f)

**Comment by epszaw** [comment]
Hey there! Will investigate the problem and give you feedback as soon as possible!

**Comment by epszaw** [comment]
In new version (1.0.7) of the extension, the problem should be solved. Please, try the latest version

**Comment by KristofKuli** [comment]
Hi @epszaw,

We have tried with the latest version, but we are still receiving error.
![image](https://github.com/allure-framework/allure2/assets/103577855/5b3fa451-d0d2-475b-9b97-838ac4918cc2)

**Reply by epszaw** [reply]
That's pretty interesting. Could you see the report after the page reload?

**Reply by KristofKuli** [reply]
No, report was still not visible. The "Loading the report" icon is spinning forever.

**Reply by KristofKuli** [reply]
Hi @epszaw.

Is there any update on this?

**Reply by epszaw** [reply]
Hey! Yes, it is. I published `1.0.8` with additional information (client and task both). Please, run the pipeline one more time and see the task logs (it would be nice if you attach them here). Also, check the report tab. I added error screen with details about the report file loading. Hope, this help us to figure out, what's going wrong.

**Reply by KristofKuli** [reply]
Hi,
Basically the same thing happens with the new version as well.
For small reports the following is displayed:
![image](https://github.com/allure-framework/allure2/assets/103577855/803b7caf-1601-464d-bea5-ed94064d8f9d)
But, if we click on the Allure Report tab, then it is still loading forever.
![image](https://github.com/allure-framework/allure2/assets/103577855/72a487e3-c453-43e6-a2a2-ea9b91af826a)
For large reports, we see the following:
![image](https://github.com/allure-framework/allure2/assets/103577855/2e2bc90a-d3ce-44ec-ace2-5906b6431b3b)
Attached the log of the task as well.
[26_Publish Allure Report.txt](https://github.com/allure-framework/allure2/files/14431316/26_Publish.Allure.Report.txt)

**Reply by epszaw** [reply]
> But, if we click on the Allure Report tab, then it is still loading forever.

This problem was solved in `1.0.9`

> For large reports, we see the following

That's very strange. What do you see in network tab in the console (filter requests by `allure.report`)? What happens with the attachment request?

**Reply by KristofKuli** [reply]
Hi,

This is what we see in network tab for allure.report.
![image](https://github.com/allure-framework/allure2/assets/103577855/54b3f88a-093c-4102-b803-11b71aa52902)

**Reply by epszaw** [reply]
That's very interesting because it can't find any allure report attachment (which actually has been attached, hasn't it?). We can have problem with attachment size limit. As you said, your report is huge. Could you tell me the report file size? Could you generate it locally?

<img width="1352" alt="image" src="https://github.com/allure-framework/allure2/assets/13414205/e306cc31-9b0c-4b8a-bb3e-d8763df77ec9">

**Reply by KristofKuli** [reply]
Locally it works without issues. The allure-report folder is around 100-130 MB.

**Reply by epszaw** [reply]
Unfortunately, [there is a size limit of 60mb for attachments in Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/organizations/settings/work/object-limits?view=azure-devops), but the extension works with a single file report. I'll try to research another way to upload attachments as soon as possible, but the current extension, unfortunately, doesn't cover your case.

**Reply by epszaw** [reply, marked as answer]
Good news! I did the fix. It's available since version `1.1.0` (the latest one at this moment).
On these screenshots, you can see difference between my 100 tests report (it's only ~10mb):

<img width="1341" alt="Screenshot 2024-02-29 at 18 36 35" src="https://github.com/allure-framework/allure2/assets/13414205/153de319-5a63-409a-a889-95c5650e893d">
 
And my 1000 tests report (divided on multiple chunks):

<img width="1346" alt="Screenshot 2024-02-29 at 18 47 50" src="https://github.com/allure-framework/allure2/assets/13414205/cd3fee92-b172-4eaf-93f5-29d40d966325">

Hope the solution will solve your problem.

**Reply by KristofKuli** [reply]
Hi Konstantin,

So far it's working. We will keep testing it today and you let you know about the result on Monday.

**Reply by KristofKuli** [reply]
Hi Konstantin,

I can confirm that extension is working properly with version 1.1.0. Thank you for the quick help!

**Reply by epszaw** [reply]
Glad to hear that! No problem :)
